gui/src/mainwindow.cpp

```diff
@@ -0,0 +1,1459 @@
+        ui->bboxWestLineEdit->setEnabled(bboxOverridden);
+
+        //local image selection
+        ui->localImageFileLabel->setEnabled(true);
+        ui->localImageFileLineEdit->setEnabled(true);
+        ui->localImageFileBrowseButton->setEnabled(true);
+
+        //token
+        ui->tokenLabel->setEnabled(false);
+        ui->tokenLineEdit->setEnabled(false);
+
+        //map id
+        ui->mapIdLabel->setEnabled(false);
+        ui->mapIdLineEdit->setEnabled(false);
+
+        //zoom
+        ui->zoomLabel->setEnabled(false);
+        ui->zoomSpinBox->setEnabled(false);
+
+        //downloads
+        ui->downloadsLabel->setEnabled(false);
+        ui->downloadsSpinBox->setEnabled(false);
+
+        //user credentials
+        ui->usernameLabel->setEnabled(false);
+        ui->usernameLineEdit->setEnabled(false);
+        ui->passwordLabel->setEnabled(false);
+        ui->passwordLineEdit->setEnabled(false);
+
+        //validate the old imagepath
+        on_imagepathLineEditLostFocus();
+    }
+}
+
+void MainWindow::on_viewMetadataButton_clicked()
+{
+    QMessageBox::information(
+        this,
+        tr("Metadata"),
+        tr("Viewing Metadata is currently not supported."));
+}
+
+void MainWindow::on_nmsCheckBox_toggled(bool checked)
+{
+    if(checked) {
+        ui->nmsSpinBox->setEnabled(true);
+    }
+    else {
+        ui->nmsSpinBox->setEnabled(false);
+    }
+}
+
+void MainWindow::on_bboxOverrideCheckBox_toggled(bool checked)
+{
+    if(checked) {
+        ui->bboxNorthLineEdit->setEnabled(true);
+        ui->bboxSouthLineEdit->setEnabled(true);
+        ui->bboxEastLineEdit->setEnabled(true);
+        ui->bboxWestLineEdit->setEnabled(true);
+    }
+    else {
+        //set bbox values back to the local image's bounding box
+        if(hasValidLocalImagePath) {
+            on_imagepathLineEditLostFocus();
+        }
+
+        ui->bboxNorthLineEdit->setEnabled(false);
+        ui->bboxSouthLineEdit->setEnabled(false);
+        ui->bboxEastLineEdit->setEnabled(false);
+        ui->bboxWestLineEdit->setEnabled(false);
+    }
+}
+
+void MainWindow::on_outputLocationBrowseButton_clicked()
+{
+    QString path = QFileDialog::getExistingDirectory(this, tr("Select Output Location"));
+    if(!path.isEmpty() && !path.isNull()) {
+        ui->outputLocationLineEdit->setText(path);
+        //manually invoke the slot to check the new directory path
+        on_outputLocationLineEditLostFocus();
+    }
+}
+
+void MainWindow::on_helpPushButton_clicked()
+{
+    QMessageBox::information(
+        this,
+        tr("Help"),
+        tr("Help is currently not supported."));
+}
+
+void MainWindow::on_runPushButton_clicked()
+{
+    QString errorBuffer;
+    ui->runPushButton->setEnabled(false);
+    bool validJob = validateUI(errorBuffer);
+
+    if(!validJob) {
+        ui->runPushButton->setEnabled(true);
+        QMessageBox::critical(
+            this,
+            tr("Error"),
```

Comment:
While I concede that my example of "descriptive of what is happening" wasn't the right path, I still think we want to give the dialog a little attention.  The major UI guides that have sections on error modal boxes (mac and windows) suggest using "use the title to identify the command or feature from which the error originated." and "no title", respectively.  They also suggest that "the main instruction [describes] the problem in clear, plain, specific language" and "an alert message that describes the alert situation clearly and succinctly."

In any case, dialog title is a bit of a wild card, but contents should suggest what went wrong (such as "Processing cannot proceed with invalid arguments.") and the details (what you have) should be in smaller text following that.

Suggest:
![image](https://cloud.githubusercontent.com/assets/7284879/23221925/54fbb4fc-f8f4-11e6-8643-79e0b3833602.png)


Current error (note the qt has a mechanism to treat the title, informative text, and detailed text in a platform-dependent way.):
![image](https://cloud.githubusercontent.com/assets/7284879/23220849/3f878248-f8f1-11e6-8a6a-973742276a59.png)

[Windows error dialog](https://msdn.microsoft.com/en-us/library/windows/desktop/dn742471.aspx):
![image](https://cloud.githubusercontent.com/assets/7284879/23220860/471ba098-f8f1-11e6-9c0d-66fbb3e2640a.png)

[Mac error dialog](https://developer.apple.com/library/content/documentation/UserExperience/Conceptual/OSXHIGuidelines/WindowAlerts.html):
![image](https://cloud.githubusercontent.com/assets/7284879/23220867/4a843e70-f8f1-11e6-9d7b-c2f601a38a78.png)